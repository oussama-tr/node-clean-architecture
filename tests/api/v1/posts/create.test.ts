import { makeClient } from '../../client';

describe('POST /api/v1/posts', () => {
  describe('given an invalid request', () => {
    it('should respond with a 400 status code', async () => {
      // Arrange
      const { client } = await makeClient();

      // Act
      const response = await client.post('/api/v1/posts').send({
        title: '', // Cannot be empty
      });

      // Assert
      expect(response.status).toEqual(400);
    });
  });

  describe('given a valid request', () => {
    it('should create post and respond with a 201 status code', async () => {
      // Arrange
      const { client } = await makeClient();

      // Act
      const createResponse = await client.post('/api/v1/posts').send({
        title: 'Example post',
      });

      // Assert
      expect(createResponse.status).toEqual(201);
      expect(createResponse.body).toMatchSnapshot({
        id: expect.stringMatching(
          /^[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}$/,
        ),
      });
    });
  });
});
