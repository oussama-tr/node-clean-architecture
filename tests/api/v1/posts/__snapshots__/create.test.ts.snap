// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POST /api/v1/posts given a valid request should create post and respond with a 201 status code 1`] = `
{
  "id": StringMatching /\\^\\[0-9a-fA-F\\]\\{8\\}\\\\b-\\[0-9a-fA-F\\]\\{4\\}\\\\b-\\[0-9a-fA-F\\]\\{4\\}\\\\b-\\[0-9a-fA-F\\]\\{4\\}\\\\b-\\[0-9a-fA-F\\]\\{12\\}\\$/,
}
`;
